codechain/rpc_apis.rs

```diff
@@ -35,6 +35,7 @@ impl ApiDependencies {
         use crpc::v1::*;
         handler.extend_with(ChainClient::new(Arc::clone(&self.client)).to_delegate());
         handler.extend_with(MempoolClient::new(Arc::clone(&self.client)).to_delegate());
+        handler.extend_with(SnapshotClient::new(Arc::clone(&self.client)).to_delegate());
```

Comment:
Why don't we just pass Config to this function? Why take a detour through `Client` to pass a `snapshot_path`?

Reply:
We can't pass the full config object since it introduces a circular dependency.
We can pass String instead, but I can't see any explicit benefit here.
Do you have any reason for changing it?

Reply:
What do you mean by a circular dependency? An `Arc` circular dependency or a module dependency?

Reply:
A module dependency.
`SnapshotClient` is in `rpc` crate and `Config` is in `codechain` crate.
Both crates need to depend on each other if we have to pass `Config` to `SnapshotClient`.

Reply:
Okay. But I still don't think punching through the client is a good idea. We put too much information into the client. We might try to separate the config module?